### ISSUE TITLE
deep check aws_launch_configuration_invalid_iam_profile fails when using assumed role

### ISSUE THREAD

**Issue by grimm26** [terraform-linters/tflint#509]
## Version
tflint version 0.12.1
terraform v0.12.13

```hcl
provider "aws" {
  region = "us-east-2"
  alias  = "oh"
  assume_role { role_arn = local.assumed_role_arn }
}
```
My ec2 related resources use that provider.  I run terraform with my unprivileged IAM user in effect, which only has the ability to assume roles.  *That* user is the user that tflint is trying to use to `ListInstanceProfiles`.  

```
> $ tflint --deep                                                                                                                                                                                                     
Failed to check `aws_launch_configuration_invalid_iam_profile` rule. An error occurred:

Error: An error occurred while invoking ListInstanceProfiles; AccessDenied: User: arn:aws:iam::12345678912:user/grimm is not authorized to perform: iam:ListInstanceProfiles on resource: arn:aws:iam::12345678912:instance-profile/
```

Shouldn't tflint use the provider that is defined in the terraform code?

**Comment by wata727** [comment]
Hi @gliptak.

Unfortunately, TFLint doesn't support `local.*` value and will ignore the assume role configuration in such cases. Please pass your AWS credentials in `.tflint.hcl`.
https://github.com/wata727/tflint/blob/v0.12.1/docs/guides/credentials.md

**Comment by grimm26** [comment]
The documentation says "TFLint can assume a role in the same way as Terraform." so I assumed it would work.

**Comment by wata727** [comment]
Certainly, this explanation may be confusing. The fact that local value does not work is documented in a separate page, which is all assumptions.
https://github.com/terraform-linters/tflint/blob/v0.12.1/docs/guides/compatibility.md#named-values

**Comment by wata727** [comment]
Close this due to no activity.
